hangman.py

```diff
@@ -0,0 +1,215 @@
+      which letters in secret_word have been guessed so far.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    pass
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    pass
+
+
+def warnings(value_mistake, warnings_point, guesses_point):
+    """
+    Subtract warnings point and output name of error
+    """
+    name_mistake = f"Oops! {value_mistake}"
+    if warnings_point > 0:
+        warnings_point -= 1
+        print(name_mistake, "\nYou have", warnings_point, "warnings left:", end=" ")
+    else:
+        guesses_point -= 1
+        print("Oops! You have no warnings left, so you lose one guess:", end=' ')
+    return warnings_point, guesses_point
+
+
+def check_star_30(count, secret_word, letters_guessed):
+    pass
+
+
+def letter_in_word(secret_word, letter, list_vowels, letter_guessed, guesses_point, letters_guessed):
+    """
+    Check if letter is word, else subtracts point
+    """
+    letters_guessed.append(letter)
+    if letter in secret_word:
+        print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+    else:
+        if letter in list_vowels:
+            guesses_point -= 2
+        else:
+            guesses_point -= 1
+        print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+    return letters_guessed, guesses_point
+
+
+def classic_input(count_star, warnings_point, guesses_point):
+    if count_star != 0:
+        warnings_point, guesses_point = warnings("you are playing classic hangman", warnings_point, guesses_point)
+    else:
+        print("OOps, you are playing classic hangman:", end=" ")
+        count_star += 1
+    return warnings_point, guesses_point, count_star
+
+
+def hangman(secret_word, request, list_vowels, warnings_point, letters_guessed, guesses_point, count_star):
+    '''
+    Game Hangman. "request" show hangman(classic) or hangman with hints.
+    Print information for user, start loop, check user choose hangman or hangman with hints,
+    Check input letter and call letter_in_word, check if user guessed word.
+    '''
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+    while guesses_point > 0:
+        print("-------")
+        print("You have", guesses_point, "guesses left")
+        print("Available letters:", end=" ")
+        get_available_letters(letters_guessed)
+        letter = input("Please guess a letter:")
+        if request == "hints" and letter == "*":
+            check_star_30(count_star, secret_word, letters_guessed)
+            count_star += 1
+        else:
+            if letter == "*":
```

Comment:
You use that "*" symbol not only once, and it has some role in the game, so I believe it makes sense to make it as a constant